File: Demo/MultilevelMenuOC/MultilevelMenuOC/ViewController.m
//
//  ViewController.m
//  MultilevelMenuOC
//
//  Created by forwor on 2019/8/9.
//  Copyright © 2019 forwor. All rights reserved.
//

#import "ViewController.h"
#import "MultilevelMenuOC-Swift.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UILabel *resultLabel;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (IBAction)testAction:(id)sender {
    NSMutableArray *dataSource = [NSMutableArray array];
    NSString *path = [[NSBundle mainBundle] pathForResource:@"businessType" ofType:@"json"];
    NSData *data = [NSData dataWithContentsOfFile:path];
    NSArray *tempArray = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:NULL];
    for (NSDictionary *dic in tempArray) {
        //将字典数组转换成模型数组
        MenuDataModel *model = [[MenuDataModel alloc] initWithDict:dic];
        //将模型存入到属性中
        [dataSource addObject:model];
    }
    
    __weak typeof(self) weakSelf = self;
    MultilevelStyle1Menu *menu = [[MultilevelStyle1Menu alloc]initWithTitle:@"行业类型" dataSouce:dataSource option:nil customView:nil completion:^(NSString * text, MenuDataModel * model) {
        weakSelf.resultLabel.text = text;
    }];
    [menu show];
}

- (IBAction)customMenu:(id)sender {
    NSString *path = [[NSBundle mainBundle] pathForResource:@"businessType" ofType:@"json"];
    NSURL *url = [NSURL fileURLWithPath:path];
    MultilevelMenuOption *option = [[MultilevelMenuOption alloc]init];
    option.rightBarButtonTitle = @"ok";
    option.rightBarButtonColor = [UIColor redColor];
    __weak typeof(self) weakSelf = self;
    MultilevelStyle2Menu *menu = [[MultilevelStyle2Menu alloc] initWithTitle:@"请选择行业类型" fileUrl:url option:option customView:nil completion:^(NSString * text, MenuDataModel * model) {
        weakSelf.resultLabel.text = text;
    }];
    [menu show];
                                                                                       
}
    

@end

File: Demo/MultilevelMenuSwift/ViewController.swift
//
//  ViewController.swift
//  MultilevelMenuDemo
//
//  Created by ChokShen on 2018/4/27.
//  Copyright © 2018年 ChokShen. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    private var customView: MultilevelMenuStyle1View!

    override func viewDidLoad() {
        super.viewDidLoad()
//        // Do any additional setup after loading the view, typically from a nib.
    }
    
    func getClassName(_ view: MultilevelMenuStyle1View) -> String {
        return String(describing: type(of: view))
    }
    
    @IBOutlet weak var resultLabel: UILabel!
    @IBAction func testAction(_ sender: Any) {
        var dataSouce: [MenuDataModel] = []
        guard let path = Bundle.main.path(forResource:"businessType", ofType: "json") else { return }
        let url = URL(fileURLWithPath: path)
        do{
            let data = try Data(contentsOf: url)
            let json:Any = try JSONSerialization.jsonObject(with: data, options:JSONSerialization.ReadingOptions.mutableContainers)
            if let jsonArray = json as? [[String: String]] {
                for dict in jsonArray {
                    let dataModel = MenuDataModel.init(dict: dict)
                    dataSouce.append(dataModel)
                }
            }
        }catch{
            print("读取本地数据出现错误！",error)
        }
        let menu = MultilevelStyle1Menu(title: "行业类型", dataSouce: dataSouce, completion: { (resultString, model) in
            self.resultLabel.text = resultString
        })
        //menu.allowSelectAnyLevelData = true
        menu.show()
    }

    @IBAction func customMenu(_ sender: Any) {
        guard let path = Bundle.main.path(forResource:"businessType", ofType: "json") else { return }
        let url = URL(fileURLWithPath: path)
        let option = MultilevelMenuOption()
        option.rightBarButtonTitle = "ok"
        option.rightBarButtonColor = UIColor.red
        let menu = MultilevelStyle2Menu(title: "请选择行业类型", fileUrl: url, option: option, customView: CustomMenuView(), completion: { (resultString, model) in
            self.resultLabel.text = resultString
        })
        menu.show()
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

